extensions/setting/common/client.mjs

```diff
@@ -0,0 +1,54 @@
+// Copyright 2021 HITCON Online Contributors
+// SPDX-License-Identifier: BSD-2-Clause
+
+import Modal from '/static/sites/game-client/ui/modal.mjs';
+import ToolbarButton from '/static/sites/game-client/ui/toolbar-button.mjs';
+
+const SETTING_DIV = 'setting-div';
+
+class SettingModal extends Modal {
```

Comment:
It would be recommended to let the SettingModal be part of the core codebase instead of being in an extension. The reason is because we're unlikely to remove it and it'll be easier for other extensions to call it.